### PR TITLE
Support abootimg building

### PR DESCRIPTION
**Pull request by Leo-Yan**
This patch set is to enable abootimg building method; if we change to use UEFI bootloader for Hikey960, UEFI using abootimg compatible booting image.

So I think the first patch can be considered merge firstly and the second patch we can postpone to merge until UEFI has been released as official booting image on Hikey960.

### CODE REVIEW

File: tools/wltests/android/create_aboot_img.sh [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+    c_info "- ANDROID_SECOND_SIZE: ${ANDROID_SECOND_SIZE}"
+    ANDROID_SECOND_SIZE="secondaddr = ${ANDROID_SECOND_SIZE}"
+fi
+
+if [ "${ANDROID_RAMDISK_OFFSET}" ]; then
+    c_info "- ANDROID_RAMDISK_OFFSET : ${ANDROID_RAMDISK_OFFSET}"
+    ANDROID_RAMDISK_OFFSET="ramdiskaddr = ${ANDROID_RAMDISK_OFFSET}"
+fi
+
+cat > "${ARTIFACTS_PATH}"/bootimg-960.cfg << EOL
```
Comment (derkling) [inline tools/wltests/android/create_aboot_img.sh]: This `bootimg-960.cfg ` seems like an hardcoded filename valid only for hikey960, isnt'it?

We probably need an additional variable in the `definitions` of each platform, to specify the name of the configuration file to generate.

File: tools/wltests/android/create_aboot_img.sh [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+################################################################################
+
+# Ensure the output folder exists
+mkdir -p $(dirname $BOOT_IMAGE) &>/dev/null
+
+set -x
+c_info "- IMAGE_DTB: $IMAGE_DTB"
+IMAGE_DTB="$(dirname $KERNEL)/Image-dtb"
+cat "${KERNEL}" "${DTB}" > "${IMAGE_DTB}"
+abootimg --create "${BOOT_IMAGE}" -k "${IMAGE_DTB}" -r "${RAMDISK}" -f "${ARTIFACTS_PATH}"/bootimg-960.cfg
```
Comment (derkling) [inline tools/wltests/android/create_aboot_img.sh]: This `abootimg` tool should be provided by this patch and be available under `tools/wltests/android`

Have a look at how to use mkbootimg here:
https://github.com/ARM-software/lisa/blob/master/tools/wltests/android/create_boot_img.sh#L75
... on the formatting side as well ;-)

File: tools/wltests/platforms/hikey960_android-4.4/definitions [hunk 3]
```diff
@@ -5,19 +5,21 @@
 
 PLATFORM_NAME="Hikey960_Android"
 
+BOOT_UEFI="${BOOT_UEFI:-YES}"
```
Comment (derkling) [inline tools/wltests/platforms/hikey960_android-4.4/definitions]: As per previous comment, this should be better defined as:
```bash
BOOTLOADER=${BOOTLOADER:-uefi}
```
at the end of the following "Boot Image Configuration" section

File: tools/wltests/android/create_aboot_img.sh (hunk 1 above)
Comment (Leo-Yan) [reply tools/wltests/android/create_aboot_img.sh]: Will add variable for the config file.

File: tools/wltests/android/create_aboot_img.sh (hunk 2 above)
Comment (Leo-Yan) [reply tools/wltests/android/create_aboot_img.sh]: abootimg can be installed by Ubuntu apt-get command, mkbooting is not official binary. So I think it's more reasonable to directly use Ubuntu official abootimg, so will add checking and give instructions for abootimg.

File: tools/wltests/platforms/hikey960_android-4.4/build_images [hunk 4]
```diff
@@ -12,9 +12,15 @@ source "${PLATFORM_PATH}/definitions"
 ./build --image --dtbs --silent; ERROR=$?
 [ $ERROR -eq 0 ] || exit $ERROR
 
-./android/create_boot_img.sh; ERROR=$?
-[ $ERROR -eq 0 ] || exit $ERROR
+if [ "x${BOOT_UEFI}"=="xYES" ]; then
```
[the remark this answers is not in the document]
Comment (Leo-Yan) [reply tools/wltests/platforms/hikey960_android-4.4/build_images]: Thanks, will do.

File: tools/wltests/android/create_aboot_img.sh (hunk 2 above)
Comment (derkling) [reply tools/wltests/android/create_aboot_img.sh]: Yes, it that case it make sense to just split out a warning if the binary is not available with a note on which packages should be installed. Thanks.

**Comment by derkling** [comment]
Hi @Leo-Yan, any chance to update this according to the above comments?
We are trying to merge as much as possible of the outstanding PR... would be nice if this one also can get it ;-)

**Comment by douglas-raillard-arm** [comment]
Are there still outstanding issues around hikey 960 + UEFI ?

**Comment by Leo-Yan** [comment]
Hi @douglas-raillard-arm ,

I think this patch set is stable and I don't see any requirement so far for UEFI/abootimg on Hikey960.  If you agree with this, let's drop this PR.

**Pull request closed without merging**